ddl/db_test.go

```diff
@@ -5725,16 +5725,16 @@ func (s *testSerialDBSuite) TestSetTiFlashReplicaForTemporaryTable(c *C) {
 
 	tk := testkit.NewTestKitWithInit(c, s.store)
 	tk.MustExec("set tidb_enable_global_temporary_table=true")
-	tk.MustExec("drop table if exists temp")
+	tk.MustExec("set tidb_enable_noop_functions = 1")
+	tk.MustExec("drop table if exists temp, temp2")
 	tk.MustExec("create global temporary table temp(id int) on commit delete rows")
+	tk.MustExec("create temporary table temp2(id int)")
 	tk.MustGetErrCode("alter table temp set tiflash replica 1", errno.ErrOptOnTemporaryTable)
-	tk.MustExec("drop table temp")
+	tk.MustGetErrCode("alter table temp2 set tiflash replica 1", errno.ErrUnsupportedDDLOperation)
+	tk.MustExec("drop table temp, temp2")
 
-	tk.MustExec("drop table if exists normal")
 	tk.MustExec("create table normal(id int)")
 	defer tk.MustExec("drop table normal")
-	tk.MustExec("alter table normal set tiflash replica 1")
-	tk.MustQuery("select REPLICA_COUNT from information_schema.tiflash_replica where table_schema='test' and table_name='normal'").Check(testkit.Rows("1"))
```

Comment:
It's a case for `create table like` should not inherit normal table's tiflash replica option when the created table is a temporary table.  I don't think you should remove it and you should add a same test for local temporary table.

You can try to exec `make failpoint-enable` before test. Without it, the case will fail